test/val/val_id_test.cpp

```diff
@@ -112,6 +114,26 @@ const char kOpenCLMemoryModel64[] = R"(
   ASSERT_EQ(expected, result);                                                 \
   spvContextDestroy(context);
 
+#define CHECK_WITH_DIAGNOSTIC(str, expected, err_substr)                       \
+  spv_diagnostic diagnostic;                                                   \
+  spv_context context = spvContextCreate(SPV_ENV_UNIVERSAL_1_0);               \
+  std::string shader = std::string(kGLSL450MemoryModel) + str;                 \
+  spv_result_t error = spvTextToBinary(context, shader.c_str(), shader.size(), \
+                                       &binary, &diagnostic);                  \
+  if (error) {                                                                 \
+    spvDiagnosticPrint(diagnostic);                                            \
+    spvDiagnosticDestroy(diagnostic);                                          \
+    ASSERT_EQ(SPV_SUCCESS, error) << shader;                                   \
+  }                                                                            \
+  spv_result_t result = spvValidate(context, get_const_binary(), &diagnostic); \
+  if (SPV_SUCCESS != result) {                                                 \
+    spvDiagnosticPrint(diagnostic);                                            \
+    EXPECT_THAT(diagnostic->error, HasSubstr(err_substr));                     \
+    spvDiagnosticDestroy(diagnostic);                                          \
+  }                                                                            \
+  ASSERT_EQ(expected, result);                                                 \
```

Comment:
Normally use EXPECT_EQ .
Use ASSERT_EQ when you check a condition that gates any further execution.  For example, use ASSERT_NE(nullptr, p)  when in following steps you dereference pointer p.  The assert prevents a crash in a test failure case.